src/condor_contrib/bosco/bosco_findplatform

```diff
@@ -53,135 +52,84 @@ def findversion_mac(detail=False):
+    return os_info
 
-Distributor ID:    Ubuntu
-Description:    Ubuntu 11.10
-Release:    11.10
-Codename:    oneiric       
-"""
-  retv=UNSUPPORTED
-  lines = open('/etc/lsb-release').readlines()
-  for line in lines:
-    if detail:
-       if 'DISTRIB_DESCRIPTION' in line:
-        return line[len('DISTRIB_DESCRIPTION='):]
-    if 'DISTRIB_ID' in line:
-      if not 'Debian' in line and not 'Ubuntu' in line:
-        return UNSUPPORTED
-    if 'DISTRIB_RELEASE' in line:
-        # TODO These checks need to to be conditioned on whether the
-        #   distro is Debian or Ubuntu.
-        if line[len('DISTRIB_RELEASE='):].startswith('9.'):
-          retv=DEB9
-        if line[len('DISTRIB_RELEASE='):].startswith('10.'):
-          retv=DEB10
-        if line[len('DISTRIB_RELEASE='):].startswith('16.'):
-          retv=UBUNTU16
-        if line[len('DISTRIB_RELEASE='):].startswith('18.'):
-          retv=UBUNTU18
-        if line[len('DISTRIB_RELEASE='):].startswith('20.'):
-          retv=UBUNTU20
-  return retv
 
-def findversion():
+def findversion() -> str:
   if not os.name=='posix':
-    return UNSUPPORTED
+    return "UNSUPPORTED"
   if sys.platform=='darwin':
     myver = platform.mac_ver()
     if myver[0]:
       if '.'.join(myver[0].split('.')[:2]) in SUPPORTED_OSX:
-        return OSX
+        return "MAC"
     return findversion_mac()
   elif sys.platform.startswith('linux'):
     # only 64 bit supported
     if not platform.architecture()[0]=='64bit':
-      return UNSUPPORTED
-    # try first platform.dist, use it only for positive recognition
-    mydist = platform.dist()
-    if mydist[0]:
-      if mydist[0].lower()=='redhat':
-        if mydist[1].startswith('6.'):
-          return RH6
-        if mydist[1].startswith('7.'):
-          return RH7
-        if mydist[1].startswith('8.'):
-          return RH8
-      if mydist[0].lower()=='debian':
-        if mydist[1].startswith('9.'):
-          return DEB9
-        if mydist[1].startswith('10.'):
-          return DEB10
-      if mydist[0].lower()=='ubuntu':
-        if mydist[1].startswith('16'):
-          return UBUNTU16
-        if mydist[1].startswith('18'):
-          return UBUNTU18
-        if mydist[1].startswith('20'):
-          return UBUNTU20
-      if mydist[0].lower()=='linuxmint':
-        if mydist[1].startswith('18'):
-          return UBUNTU16
-        if mydist[1].startswith('19'):
-          return UBUNTU18
-        if mydist[1].startswith('20'):
-          return UBUNTU20
-    if os.path.isfile('/etc/redhat-release'):
-      return findversion_redhat()
-    elif os.path.isfile('/etc/lsb-release'):
-      return findversion_debian()
-  return UNKNOWN
+      return "UNSUPPORTED"
+
+    try:
+      distro_info = get_distro_info()
+    except FileNotFoundError:
+      return "UNKNOWN"
+
+    distro = distro_info['ID']
+    bosco_distro = DISTRO_MAPPING[distro]
+    major_ver = distro_info['VERSION_ID'].split('.')[0]
+
+    if distro == 'linuxmunt':
```

Comment:
```suggestion
    if distro == 'linuxmint':
```

Reply:
You haven't heard of the popular distro Linux Munt?